### ISSUE TITLE
build failed

### ISSUE THREAD

**Issue by mintty** [mintty/wsltty.appx#6]
Concerning https://github.com/mintty/wsltty/issues/63#issuecomment-426788592:
After building wslbridge and mintty, makefile invokes build.sh with the following output, with some intermediate popups where I entered certificate passwords; sorry, MS tool messages are in German (unaffected by LC_ALL). I may have to update my tool set for the current Windows version (10, 64 bit, updated yesterday).

cleaning old build artefacts
copying assets
Output location of PRI file(s) is detected to be under the project root. If previously generated PRI file(s) are present, these may be re-indexed and lead to unexpected results.
Index Pass Completed.
AlternateForm Qualifiers: UNPLATED
Scale Qualifiers: 100,125,150,200,400
TargetSize Qualifiers: 256,16,24,32,48

Finished building
Version: 1.0
Resource Map Name: mintty.wsltty
Named Resources: 5
Resource Candidates: 31
Successfully Completed
copying mintty
C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(1126,5): error MSB3644: Die Verweisassemblys f�r Framework ".NETFramework,Version=v4.6.1" wurden nicht gefunden. Installieren Sie zum Beheben dieses Problems das SDK oder das Paket zur Festlegung vonZielversionen f�r die vorliegende Frameworkversion, oder legen Sie als neues Ziel f�r die Anwendung eine Version des Frameworks fest, f�r die Sie das SDK oder Paket zur Festlegung der Zielversionen installiert haben. Assemblys werden im globalen Assemblycache (GAC) aufgel�st und anstelle von Verweisassemblys verwendet. Daher wird die Assembly f�r das gew�nschte Framework unter Umst�nden nicht ordnungsgem�� als Ziel festgelegt. [C:\cygwin\home\towo\mintty\wsltty.appx\Launcher\Launcher.csproj]
/bin/cp: cannot stat 'Launcher/Launcher.exe': No such file or directory
/bin/cp: cannot stat 'Launcher/Launcher.exe.config': No such file or directory
Microsoft (R) MakeAppx Tool
Copyright (C) 2013 Microsoft.  All rights reserved.

The path (/p) parameter is: "\\?\C:\cygwin\home\towo\mintty\wsltty.appx\wsltty_unsigned.appx"
The content directory (/d) parameter is: "\\?\C:\cygwin\home\towo\mintty\wsltty.appx\build"
Enumerating files from directory "\\?\C:\cygwin\home\towo\mintty\wsltty.appx\build"
Packing 56 file(s) in "\\?\C:\cygwin\home\towo\mintty\wsltty.appx\build" (content directory) to "\\?\C:\cygwin\home\towo\mintty\wsltty.appx\wsltty_unsigned.appx" (output file name).
Memory limit defaulting to 4240934912 bytes.
Using "\\?\C:\cygwin\home\towo\mintty\wsltty.appx\build\AppXManifest.xml" as the manifest for the package.
MakeAppx : error: Manifest validation error: Line 5, Column 18, Reason: The file name "bin\launcher.exe" declared for element "*[local-name()='Applications']/*[local-name()='Application']" doesn't exist in the package.
MakeAppx : error: Package creation failed.
MakeAppx : error: 0x80080204 - The specified package format is not valid: The package manifest is not valid.

**Comment by Biswa96** [comment]
Which Windows SDK version did you installed? Can you detect the .NET version?
* Possible solution: Open the launcher folder. Open `app.config` and `Launcher.csproj` in any text editor. Change **v4.6.1** to **v4.7.2** or **v4.7.1**.

**Comment by mintty** [comment]
.net 4.7, 4.7.1 and 4.7.2 SDKs are installed
Windows SDKs 10.0.16299.15 and 10.0.17134.12 are installed

Among the tools needed, MSBuild.exe is not found. But I don't see an error message relating to that.

**Comment by mintty** [comment]
Found MSBuild.exe somewhere else and added to path. No success.
Changed version references within Launcher. wsltty.appx is generated.
However, Windows refuses to open wsltty.appx...

**Comment by Biswa96** [comment]
Is wsltty.appx installed? Does install.bat work? Try to remove old SDK version.

**Comment by mintty** [comment]
"The package could not be opened" - not even as far as checking for old versions... Removed anyway, no change.
install.bat - you mean, from the plain wsltty package? That installs just fine.

**Comment by Biswa96** [comment]
I'm little confused. I think you solved the build faliure, and now you've problem to install it. Am I right?

I mean remove old wsltty.appx and old Windows SDK installations. Delete the wsltty.appx project folder. Start from beginning. Edit those two files as I said. Change **v4.6.1** to **v4.7.2**.Place bin and usr folder in build folder. Run the `install_appx.cmd` file. There should be only one MSbuild.exe in `where` command after running `build.cmd` file.

Does the error "Package could not be opened" shows after running the CMD file? Or you install it with double click on appx package?

**Comment by mintty** [comment]
>  I think you solved the build faliure, and now you've problem to install it. Am I right?

Yes (and the missing MSBuild was a mistake).

> Change v4.6.1 to v4.7.2.

We should try to detect the version of the installed SDK automatically and insert that, by the way...

> Does the error "Package could not be opened" shows after running the CMD file?

No, actually, I don't run that file at all. I use the makefile which invokes build.sh. Should have the same result, I'll check.

> Or you install it with double click on appx package?

Yes.

**Comment by Biswa96** [comment]
> We should try to detect the version of the installed SDK automatically and insert that, by the way...

That's why I choose C/C++ instead of C# :wink: One C file would be enough.

Forgive me, I'm not expert of shell script and makefile. I don't know if the error in those two file.

**Comment by mintty** [comment]
Using the install_appx.cmd script, I'm getting some error output telling me that the package is already installed and installing again is blocked. It suggests to increase the version number of the package, which is probably a good idea, or remove the old package first.
However, there is no old package as I can see, after I had uninstalled yesterday. No wsltty in system properties "Apps & Features", running install_appx.cmd with answer "u" fails miraculously (cannot find jump label "Uninstall" although it's there), even manually ran the line "Powershell Get-AppxPackage *wsltty* ^| Remove-AppxPackage" (no echo, no message). Puzzled.
Maybe you can suggest how to increase the package version number, for a start?

**Comment by Biswa96** [comment]
It's weird. Uninstall option should remove the package. Did you check `ProgramFiles\WindowsApps` folder if `mintty.wsltty` folder present?

The version number can be increased in `AppxManifest.xml` file. It's in the 3rd line in `<Identity>` tag.

If you've time you can experiment in a freshly installed Virtual Machine. I do that (VM snapshots) all the time so that older issues don't interfare with new builds.

**Comment by mintty** [comment]
WindowsApps/mintty.wsltty... still existed (although I had previously uninstalled via Start Menu).
After removing that, it works (with the install script). Still puzzling:
* how to improve the package to get the folder removed on uninstallation
* why running the package directly does not work

Anyway, I'll build another appx package later (with increased version number), try to adapt the SDK version automatically, and make it a release.

**Comment by Biswa96** [comment]
> how to improve the package to get the folder removed on uninstallation

It should be managed by the Windows OS itself. One of the main features of Appx package (UWP) is that it automatically removes junks.

> why running the package directly does not work

That's why I make the install.cmd script :wink:  I'll check if it has any issue in latest Windows RS5.

To build the Appx package, if you want I can provide an Appveyor CI yml file. SDK version can be automatically detected, I know (will find) the tricks.

**Comment by mintty** [comment]
Patched version handling and fixed the uninstall issue.
The package can be built and installed with the script.
But it does not run! Window flashes and terminates immediately.

**Comment by mintty** [comment]
Setting Hold=always in $APPDATA/mintty/config reveals:
C:\Program Files\WindowsApps\mintty.wsltty_1.9.3.0_x64__g625rz8yt6a9c\bin\wslbridge.exe: Exit 127.

**Comment by Biswa96** [comment]
Any update for this issue? I don't see any error in Windows 10 version 1809 (or in newer Insider builds).

**Comment by mintty** [comment]
No. If you provide your package (upload or email), and I can install it, I'll make a release. Although I'd certainly prefer to find out why my build fails to run. Do you use the makefile from the repository for building?

**Comment by Biswa96** [comment]
Do you accept packages from Appveyor CI? I shall make a test repository for that packaging only.

**Comment by Biswa96** [comment]
Do you accept packages from Appveyor CI? I shall make a test repository for that packaging only. I didn't try with makefile.

**Comment by mintty** [comment]
I'll check that out. Which version of wslbridge is used in your package? (Maybe something's broken in the newest one.)
Please provide a link to that package. If it's stored there permanently, we could also just provide a link.

**Comment by Biswa96** [comment]
Links:
Repository: https://github.com/Biswa96/wsltty_test
Appveyor CI project: https://ci.appveyor.com/project/Biswa96/wsltty-test

* Ubuntu Image > Artifacts > wslbridge-backend
* Visual Studio 2017 image > Artifacts > Unsigned Appx package

The artifacts only available for 6 months. So I will not recommend to add the direct links of Appveyor. Users should download from GitHub release page. You've to just download the unsigned Appx file and sign it with `sign_appx.cmd` and upload it in GitHub release page. All binaries are build from sources' latest commit. Check the SAH256 of Appx file before downloading from Appveyor CI.

**Comment by Biswa96** [comment]
Does this work?

**Comment by mintty** [comment]
Haven't yet had time to test...
